File: 151002/Rusakovich/NewsDistributedSystem/publisher/src/main/java/by/rusakovich/publisher/controller/AuthorController.java
package by.rusakovich.publisher.controller;

import by.rusakovich.publisher.model.dto.author.AuthorRequestTO;
import by.rusakovich.publisher.model.dto.author.AuthorResponseTO;
import by.rusakovich.publisher.service.impl.AuthorService;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("${api.url}/authors")
public class AuthorController extends Controller<Long, AuthorRequestTO, AuthorResponseTO, AuthorService> {
    public AuthorController(AuthorService service) {
        super(service);
    }
}

File: 151003/Halai/Publisher/Controllers/V1_0/AuthorController.cs
using AutoMapper;
using Microsoft.AspNetCore.Mvc;
using Publisher.Controllers.V1_0.Common;
using Publisher.Entity.Db;
using Publisher.Entity.DTO.RequestTO;
using Publisher.Entity.DTO.ResponseTO;
using Publisher.Service.Interface;
using System.Net;

namespace Publisher.Controllers.V1_0
{
    [Route("api/v1.0/authors")]
    [ApiController]
    public class AuthorController(IAuthorService AuthorService, ILogger<AuthorController> Logger, IMapper Mapper) :
        AbstractController<Author, AuthorRequestTO, AuthorResponseTO>(AuthorService, Logger, Mapper)
    {
        [HttpGet]
        [Route("news/{id:int}")]
        public async Task<JsonResult> GetByNewsID([FromRoute] int id)
        {
            AuthorResponseTO? response = null;
            Logger.LogInformation("Getting author by news ID: {id}", id);

            try
            {
                response = await AuthorService.GetByNewsID(id);
            }
            catch (Exception ex)
            {
                Logger.LogError("Error getting author by news ID {ex}", ex);
                Response.StatusCode = (int)HttpStatusCode.BadRequest;
            }

            return Json(response);
        }
    }
}
